### PR TITLE
Relation details

### PR DESCRIPTION
**Pull request by wyy1123**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ def relation():
 def relations_by_user_id(user_id):
     return RelationsView.get_relations(user_id)
 
+@app.route('/relation/<string:user_id>/<string:relation_id>', methods=['GET'])
+def relation_by_relation_id(user_id, relation_id):
```
Comment (Mark-Jung) [inline app.py]: why do you require user_id if you are not using it?

File: models/relations.py [hunk 2]
```diff
@@ -46,6 +46,15 @@ def valid_construction(cls, user_id, first, last, death_year, is_deceased, gende
 
     
     def json(self):
+        return {
+            "id": self.id,
```
Comment (Mark-Jung) [inline models/relations.py:50]: what's the difference between `json()` and `individual_json()`

File: views/relations.py [hunk 3]
```diff
@@ -28,6 +28,13 @@ def get_relations(cls, user_id):
         if error_message:
             return json.dumps({"response": error_message}), status
         return json.dumps({"response": list(map(lambda x: x.json() if x else None, response))}), status
+     
+    @classmethod
+    def get_specific_relation(cls, relation_id):
+        error_message, status, response = RelationsController.get_specific_relation(relation_id)
+        if error_message:
+            return json.dumps({"response": error_message}), status
+        return json.dumps({"response": list(map(lambda x: x.individual_json() if x else None, response))}), status
```
Comment (Mark-Jung) [inline views/relations.py]: why are you using map to iterate? There should be only one relation object that's related to one relation_id, as it is a unique field. Also, you don't even use `json()` here, which you created in this pr.

File: tests/test_relations.py [hunk 4]
```diff
@@ -242,6 +255,8 @@ def test_relation_calculation(self):
         self.assertEqual(new_relation.status_code, 201)
         get_relation = self.app.get('/relation/1')
         self.assertEqual(get_relation.status_code, 200)
+        get_specific_relation = self.app.get('/relation/1/6')
+        self.assertEqual(get_specific_relation.status_code, 200)
```
Comment (Mark-Jung) [inline tests/test_relations.py:302]: Sure, you did positive testing. (returns expected positive response) You also need to do testing for negative responses. (ex. what happens when someone requests for a relation with a relation_id that doesn't exist?)

**Review by Mark-Jung — CHANGES_REQUESTED** [review]
Overall, poor code quality. Try to be more deliberate about the code you write, as most of the code are copy pastas.

File: models/relations.py (hunk 2 above)
Comment (wyy1123) [reply models/relations.py:50]: json() is for returning list of relations for a user, while individual_json() is for returning all the parameters of a single relation. Is there a way to use the same function to return different parameters?

Comment (Mark-Jung) [reply models/relations.py:50]: I can't immediately see how `json()` would return a list. You do realize these functions share the same lines of code? Both `json()` and `individual_json()` return a dictionary.

File: views/relations.py (hunk 3 above)
Comment (Mark-Jung) [reply views/relations.py]: heard from Yul Hee, seems like your fix for this hasn't been pushed yet.

**Review by Mark-Jung — CHANGES_REQUESTED** [review]

File: models/relations.py [hunk 5]
```diff
@@ -60,10 +69,13 @@ def json(self):
...
             "nickname": self.nickname
-        }
+        }   
 
 
     @classmethod
     def find_by_user_id(cls, user_id):
         return cls.query.filter_by(user_id = user_id).all()
 
+    @classmethod
```
Comment (Mark-Jung) [inline models/relations.py]: unnecessary function. does the same job as find_by_id, which should have been inherited from the parent function, CommonModel

File: views/relations.py [hunk 6]
```diff
@@ -28,6 +28,13 @@ def get_relations(cls, user_id):
         if error_message:
             return json.dumps({"response": error_message}), status
         return json.dumps({"response": list(map(lambda x: x.json() if x else None, response))}), status
+     
+    @classmethod
+    def get_specific_relation(cls, relation_id):
+        error_message, status, response = RelationsController.get_specific_relation(relation_id)
+        if error_message:
+            return json.dumps({"response": error_message}), status
+        return json.dumps({"response": list(response.individual_json())}), status
```
Comment (Mark-Jung) [inline views/relations.py]: don't need a list. just do `response.individual_json()`

**Review by Mark-Jung — CHANGES_REQUESTED** [review]
